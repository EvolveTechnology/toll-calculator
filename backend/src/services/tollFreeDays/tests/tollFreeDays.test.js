import isTollFreeDate from '..';
import { weekends, holidays } from './mock';

const asUTC = ({ month, day, year }) => new Date(Date.UTC(year, month, day));
const asDates = arr => arr.map(asUTC);

describe('tollFreeDays', () => {
  it('labels specified holidays as tollfree', () => {
    expect(
      asDates(holidays).map(date => ({ isTollFree: isTollFreeDate(date), date })),
    ).toMatchSnapshot();
  });

  it('labels weekends as tollfree', () => {
    expect(
      asDates(weekends).map(date => ({ isTollFree: isTollFreeDate(date), date })),
    ).toMatchSnapshot();
  });
});
