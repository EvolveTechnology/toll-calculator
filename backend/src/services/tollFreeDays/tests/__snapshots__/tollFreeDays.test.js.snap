// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tollFreeDays labels specified holidays as tollfree 1`] = `
Array [
  Object {
    "date": 2013-01-01T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-03-28T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-03-29T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-04-01T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-04-30T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-05-01T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-05-08T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-05-09T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-06-05T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-06-06T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-06-21T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-07-15T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-11-01T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-12-24T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-12-25T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-12-26T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-12-31T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-09-12T00:00:00.000Z,
    "isTollFree": false,
  },
  Object {
    "date": 2013-10-17T00:00:00.000Z,
    "isTollFree": false,
  },
]
`;

exports[`tollFreeDays labels weekends as tollfree 1`] = `
Array [
  Object {
    "date": 2013-09-14T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-09-15T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-09-21T00:00:00.000Z,
    "isTollFree": true,
  },
  Object {
    "date": 2013-09-22T00:00:00.000Z,
    "isTollFree": true,
  },
]
`;
